### PR TITLE
[STORM-4030] Dependency upgrades

### PR DESCRIPTION
**Pull request by jnioche**
## What is the purpose of the change

Update dependencies

## How was the change tested

`mvn clean test` ran successfully

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -97,31 +97,31 @@
...
+        <log4j.version>2.23.0</log4j.version>
+        <slf4j.version>2.0.12</slf4j.version>
         <metrics.version>3.2.6</metrics.version>
         <mockito.version>4.11.0</mockito.version>
         <zookeeper.version>3.9.1</zookeeper.version>
         <snappy.version>1.1.10.4</snappy.version>
         <jline.version>0.9.94</jline.version>
         <hive.version>3.1.3</hive.version>
-        <hadoop.version>3.3.6</hadoop.version>
+        <hadoop.version>3.4.0</hadoop.version>
```
Comment (rzo1) [inline pom.xml:117]: Looks like `3.4.0` purged `hadoop-auth` ? Isn't listed in https://mvnrepository.com/artifact/org.apache.hadoop/hadoop-auth anymore, so might be gone and needs migration?

File: pom.xml [hunk 2]
```diff
@@ -97,31 +97,31 @@
...
-        <guava.version>32.1.3-jre</guava.version>
+        <guava.version>33.0.0-jre</guava.version>
         <auto-service.version>1.0-rc4</auto-service.version>
-        <netty.version>4.1.100.Final</netty.version>
+        <netty.version>4.1.107.Final</netty.version>
         <sysout-over-slf4j.version>1.0.2</sysout-over-slf4j.version>
-        <log4j.version>2.21.1</log4j.version>
-        <slf4j.version>1.7.36</slf4j.version>
+        <log4j.version>2.23.0</log4j.version>
+        <slf4j.version>2.0.12</slf4j.version>
```
Comment (rzo1) [inline pom.xml]: slf4j -> slf4j2 might require some changes in the log4j2 bindings used as the way the static logger bindings are loaded has changed in slf4j2.

File: pom.xml (hunk 1 above)
Comment (jnioche) [reply pom.xml]: The changes for 3.4.0 are not documented on https://hadoop.apache.org/docs/stable/ 
Will revert for now, we can work out whether hadoop-auth is needed later

Comment (jnioche) [reply pom.xml]: looks like it is still at the RC level
https://github.com/apache/hadoop/tags

**Review by rzo1 — APPROVED** [review]
Can we document the updated versions in the related Jira? 
Otherwise, it might be difficult for consumers to actually see the diff in the next release.

**Comment by jnioche** [comment]
> Can we document the updated versions in the related Jira? Otherwise, it might be difficult for consumers to actually see the diff in the next release.

done, see https://issues.apache.org/jira/browse/STORM-4030

**Comment by rzo1** [comment]
awesome

**Pull request merged**